chameleon/src/test/java/org/hibernate/omm/crud/SimpleCRUDTests.java

```diff
@@ -30,21 +30,23 @@ void tearDown() {
 	}
 
 	Book insertBook() {
-		return getSessionFactory().fromSession( session -> {
-			var book = new Book();
-			book.id = id;
-			book.title = "War and Peace";
-			book.author = "Leo Tolstoy";
-			book.publishYear = 1869;
-			session.persist( book );
-			return book;
-		} );
+		var book = new Book();
+		book.id = id;
+		book.title = "War and Peace";
+		book.author = "Leo Tolstoy";
+		book.publishYear = 1869;
+
+		getSessionFactory().inTransaction( session -> session.persist( book ));
```

Comment:
When I ran this it didn't persist when not in a txn

Reply:
the issue has been solved by starting ClientSession's transaction whenever Connection's setAutoCommit(false) as per Hibernate's logic.